### PR TITLE
feat(nextra-theme-docs): bump `rehype-pretty-code` version, support `showLineNumbers`

### PR DESCRIPTION
**Pull request by B2o5T**
related https://github.com/shuding/nextra/pull/501

### CODE REVIEW

File: examples/docs/src/pages/nextra/image.md [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
 with Nextra, is equivalent to
 
-```jsx
+```mdx
```
Comment (B2o5T) [inline examples/docs/src/pages/nextra/image.md:9]: should be `mdx` not `jsx`

File: packages/nextra-theme-docs/src/styles.css [hunk 2]
```diff
@@ -444,16 +444,30 @@ article {
...
+      }
+    }
+
     code {
-      line-height: 1.25rem;
-      @apply relative p-0 text-sm text-current bg-transparent dark:bg-transparent rounded-none border-none inline-block min-w-full;
+      @apply p-0 text-sm text-current bg-transparent dark:bg-transparent rounded-none border-none grid min-w-full leading-5;
+      .line {
+        @apply px-4 border-l-2 border-transparent;
+      }
```
Comment (B2o5T) [inline packages/nextra-theme-docs/src/styles.css:468]:
```suggestion
      }
      /* code blocks without languages don't have children that have px-4 */
      &:not([data-language]) {
        @apply px-4;
      }
```

**Comment by B2o5T** [comment]
@shuding https://nextra-theme-docs-d65itsonc.vercel.sh/docs/advanced/code-highlighting

**Comment by numToStr** [comment]
@B2o5T For blog-theme, I assume we only need to add styles in blog-theme for this to work?

**Comment by B2o5T** [comment]
@numToStr after this PR will be merged, I want to merge the following PR https://github.com/shuding/nextra/pull/534 and I'll add another PR to extract code-styles and reuse them in blog and docs

**Comment by numToStr** [comment]
@B2o5T That make sense! I'll will be easier to implement something like #539 in the future. Thanks for your all hardwork ❤️

**Review by shuding — APPROVED** [review]
Looks great, thank you!

**Pull request merged**